frontend/src/metabase/ui/components/inputs/Switch/Switch.styled.tsx

```diff
@@ -74,8 +74,10 @@ export const getSwitchOverrides = (): MantineThemeOverride["components"] => ({
           color: theme.fn.themeColor("text-dark"),
           cursor: "pointer",
           "&[data-disabled]": {
-            color: theme.fn.themeColor("text-light"),
-            cursor: "default",
+            "&:not([data-css-specificity-hack='😢'])": {
```

Comment:
as much as I love humor, as a de facto standard, we'd just double the selector that's already there for this win the specificity war.
```js
          "&[data-disabled][data-disabled]": {
            color: theme.fn.themeColor("text-light"),
            cursor: "default",
          },
```

![image](https://github.com/user-attachments/assets/b6aec457-1fa3-4034-8b74-67f24e551880)

Reply:
I disagree - adding the double` data-disabled` is confusing as to why you do it. `data-css-specificity-hack` tells you why this curious property is there - and makes it easy to remove in the future.

Reply:
If we ended up deciding to go with this at least you could combine the selector

```ts
"&[data-disabled]:not([data-css-specificity-hack='😢'])" {
xxx
```

This will prevent anyone from adding more rules directly under `"&[data-disabled]"` which could still lose to the specificity war.